Docs/reference/content/reference/driver/crud/client_side_encryption.md

```diff
@@ -181,4 +181,153 @@ namespace MongoDB.Driver.Examples
+using System.Threading;
+using MongoDB.Driver.Encryption;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver.Examples
+{
+    public class ExplicitEncryptionExamples
+    {
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+        public static void Main(string[] args)
+        {
+            var localMasterKey = Convert.FromBase64String(LocalMasterKey);
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localKey = new Dictionary<string, object>
+            {
+                { "key", localMasterKey }
+            };
+            kmsProviders.Add("local", localKey);
+
+            var keyVaultNamespace = CollectionNamespace.FromFullName("admin.datakeys");
+            var keyVaultClient = new MongoClient();
+            var keyVaultDatabase = keyVaultClient.GetDatabase(keyVaultNamespace.DatabaseNamespace.DatabaseName);
+            keyVaultDatabase.DropCollection(keyVaultNamespace.CollectionName);
+
+            // Create the ClientEncryption instance
+            var clientEncryptionSettings = new ClientEncryptionOptions(
+                keyVaultClient,
+                keyVaultNamespace,
+                kmsProviders);
+            var clientEncryption = new ClientEncryption(clientEncryptionSettings);
+
+            var dataKeyId = clientEncryption.CreateDataKey(
+                "local",
+                new DataKeyOptions(),
+                CancellationToken.None);
+
+            var originalString = "123456789";
+            Console.WriteLine($"Original string {originalString}.");
+
+            // Explicitly encrypt a field
+            var encryptOptions = new EncryptOptions(
+                EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic.ToString(),
+                keyId: dataKeyId);
+            var encryptedFieldValue = clientEncryption.Encrypt(
+                originalString,
+                encryptOptions,
+                CancellationToken.None);
+            Console.WriteLine($"Encrypted value {encryptedFieldValue}.");
+
+            // Explicitly decrypt the field
+            var decryptedValue = clientEncryption.Decrypt(encryptedFieldValue, CancellationToken.None);
+            Console.WriteLine($"Decrypted value {decryptedValue}.");
+
+            clientEncryption.Dispose();
+        }
+    }
+}
+```
+
+### Explicit Encryption and Auto Decryption
+
+Although automatic encryption requires MongoDB 4.2 enterprise or a MongoDB 4.2 Atlas cluster, automatic decryption is supported for all users. To configure automatic decryption without automatic encryption set `bypassAutoEncryption=true`. The following example has been adapted from [`ExplicitEncryptionExamples.cs`](https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Examples/ExplicitEncryptionExamples.cs):
```

Comment:
Capitalize `Enterprise`?

Do we really need the link to `ExplicitEncryptionExamples.cs`?

Reply:
>Do we really need the link to ExplicitEncryptionExamples.cs?

This is how it was implemented before (in the first version of FLE doc). Java also has a similar link (but not the same). Since the code which we provide can be used by a user without any changes(just copypaste), I think this link is not required, but if we want to remove it, it should be removed from the previous FLE version too

Reply:
I guess it doesn't matter to me whether the link is present or not.

It just seems to be both

* duplication: because the code linked to is almost the same as this
* confusing: because the code linked to is not exactly the same as this

Reply:
>confusing: because the code linked to is not exactly the same as this

that's why we used "`adapt`" word. Since java also has links on the code in the driver, I would suggest to leave it